### PR TITLE
Fix sentinel token IDs in data collator for Flax T5 pretraining script

### LINKED ISSUE #14282: Mismatch between sentinel token IDs from T5 data collator and T5 tokenizer
## Environment info

- `transformers` version: 4.10.3
- Platform: Linux-4.18.0-240.el8.x86_64-x86_64-with-glibc2.28
- Python version: 3.9.7
- PyTorch version (GPU?): 1.10.0 (True)
- Tensorflow version (GPU?): 2.6.0 (True)
- Flax version (CPU?/GPU?/TPU?): 0.3.5 (gpu)
- Jax version: 0.2.24
- JaxLib version: 0.1.73
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: No

### Who can help

@patil-suraj @patrickvonplaten

## Information

I'm trying to use the [`run_t5_mlm_flax.py`](https://github.com/huggingface/transformers/blob/7db2a79b387fd862ffb0af72f7148e6371339c7f/examples/flax/language-modeling/run_t5_mlm_flax.py) script to do additional pretraining of T5, and I noticed something strange about the way the data collator adds mask/sentinel tokens. In [line 293](https://github.com/huggingface/transformers/blob/7db2a79b387fd862ffb0af72f7148e6371339c7f/examples/flax/language-modeling/run_t5_mlm_flax.py#L293) of `run_t5_mlm_flax.py`, the `create_sentinel_ids` function replaces the masked positions with the corresponding sentinel IDs as `sentinel_ids + self.tokenizer.vocab_size - 1`, which gives values of `32100, 32101, 32102, ...`. However, the sentinel tokens `<extra_id_0>, <extra_id_1>, <extra_id_2>, ...` in the tokenizer for `t5-base` have the token IDs `32099, 32098, 32097, ...`, which I'm getting from running the following:

```python
from transformers import AutoTokenizer
tokenizer = AutoTokenizer.from_pretrained('t5-base')
print(tokenizer.convert_tokens_to_ids(['<extra_id_0>', '<extra_id_1>', '<extra_id_2>', '<extra_id_99>']))
# prints:
# [32099, 32098, 32097, 32000]
```

The larger token IDs seem to work without error because the `T5ForConditionalGeneration` pretrained model has an extra 128 token embeddings (even though `tokenizer.vocab` gives a value of `32100`, which seems to be related to issue #4875), but I'm not sure if these are the same embeddings that were used for the sentinel tokens during the original pretraining. Is the script correct in replacing the mask tokens with token IDs starting from `32100`, even though they don't correspond to the `<extra_id_#>` tokens in the vocabulary?

Here's an example of the behavior of `create_sentinel_ids` alone:

```python
import numpy as np
from transformers import AutoTokenizer

def create_sentinel_ids(mask_indices, tokenizer):
    start_indices = mask_indices - np.roll(mask_indices, 1, axis=-1) * mask_indices
    start_indices[:, 0] = mask_indices[:, 0]
    sentinel_ids = np.where(start_indices != 0, np.cumsum(start_indices, axis=-1), start_indices)
    sentinel_ids = np.where(sentinel_ids != 0, (sentinel_ids + tokenizer.vocab_size - 1), 0)
    sentinel_ids -= mask_indices - start_indices
    return sentinel_ids

tokenizer = AutoTokenizer.from_pretrained('t5-base')
mask_indices = np.array([[0, 1, 0, 0, 1, 1, 0, 1, 1, 1, 0, 1]]).astype(bool)
print(create_sentinel_ids(mask_indices.astype(np.int8), tokenizer))
# prints:
# [[    0 32100     0     0 32101    -1     0 32102    -1    -1     0 32103]]
```

### PR DESCRIPTION
**Pull request by rahuln**
# What does this PR do?

Modifies the sentinel token IDs used in the data collator for the Flax T5 pretraining script so that they go in decreasing order starting at `len(tokenizer) - 1`, which matches the [original T5 code](https://github.com/google-research/text-to-text-transfer-transformer/blob/main/t5/data/preprocessors.py#L2895).

Fixes #14282 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by LysandreJik** [comment]
Hey @rahuln! I'm pinging @patrickvonplaten for review as you have worked with him until now, but please note that he's off until next week so he'll review your PR when he's back! Thanks for your understanding.

**Comment by patrickvonplaten** [comment]
Great! Thanks a lot for digging into this issue and fixing it

File: examples/flax/language-modeling/run_t5_mlm_flax.py [hunk 1]
```diff
@@ -290,7 +290,7 @@ def create_sentinel_ids(self, mask_indices):
         start_indices[:, 0] = mask_indices[:, 0]
 
         sentinel_ids = np.where(start_indices != 0, np.cumsum(start_indices, axis=-1), start_indices)
```
Comment (thomasw21) [inline examples/flax/language-modeling/run_t5_mlm_flax.py:292]: Is that line not completely unecessary since we have the line right after?

Comment (patrickvonplaten) [reply examples/flax/language-modeling/run_t5_mlm_flax.py:292]: Well the next line makes use of the just changed `sentinel_ids` parrameter no?

Comment (thomasw21) [reply examples/flax/language-modeling/run_t5_mlm_flax.py:292]: Sorry what I meant is those two lines should be summarizable to 
```python
sentinel_ids = np.where(start_indices != 0, (len(self.tokenizer) - np.cumsum(start_indices, axis=-1)), 0)
```
Ie what is 0 is kept at 0 and what's not 0 is given a non 0 value, which means that the next `where` operation uses the same segmentation and thus overrides the values.

cc @patil-suraj
